src/hls-adapter.js

```diff
@@ -1080,6 +1093,11 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
     const errorFatal = data.fatal;
     let errorDataObject = this._getErrorDataObject(data);
     if (errorFatal) {
+      if (this._loadPromiseHandlers) {
```

Comment:
unnecessary. the promise will rejected by destroy if it's critical error. if not it should be pending

Reply:
done